federation/pkg/federation-controller/sync/controller.go

```diff
@@ -438,10 +450,32 @@ func syncToClusters(clustersAccessor clustersAccessorFunc, operationsAccessor op
 		return statusNotSynced
 	}
 
-	operations, err := operationsAccessor(adapter, clusters, obj)
+	var userInfo interface{}
+	if objUpdatingAdapter, ok := adapter.(federatedtypes.ObjectUpdatingAdapter); ok {
+		userInfo, err = objUpdatingAdapter.PrepareForUpdate(obj, key, clusters, informer)
```

Comment:
I see one problem here when used with cluster selector.
The schedule is called before the cluster selector logic over all clusters in the list and then those clusters(with the result) are discarded which do not get selected by cluster selector. The over all schedule is thus wrong. The overall schedule logic should work on only those clusters which get selected by cluster selector.